packages/components/src/templates/next/components/complex/ImageGallery/ImageGalleryClient.tsx

```diff
@@ -0,0 +1,295 @@
+"use client"
+
+import { useCallback, useMemo, useRef, useState } from "react"
+
+import type { ImageGalleryClientProps } from "~/interfaces/complex/ImageGallery"
+import { useBreakpoint } from "~/hooks/useBreakpoint"
+import { tv } from "~/lib/tv"
+import { ImageClient } from "../Image/ImageClient"
+import { LEFT_ARROW_SVG, RIGHT_ARROW_SVG } from "./constants"
+import { getEndingPreviewIndices, getPreviewIndices } from "./utils"
+
+const TRANSITION_DURATION = 150
+
+const createImagePreviewStyles = tv({
+  slots: {
+    container:
+      "relative aspect-[1/1] w-full flex-1 flex-shrink-0 overflow-hidden border-[1px] focus-visible:outline focus-visible:outline-[0.75rem] focus-visible:outline-offset-[-0.75rem] focus-visible:outline-utility-highlight",
+  },
+  variants: {
+    isSelected: {
+      true: {
+        container:
+          "border-base-content outline outline-[0.25rem] outline-offset-[-0.25rem] outline-base-content",
+      },
+      false: {
+        container: "border-base-divider-subtle hover:opacity-80",
+      },
+    },
+    numberOfImages: {
+      "3": {
+        container: "h-[7.375rem]",
+      },
+      "5": {
+        container: "h-[5.375rem]",
+      },
+    },
+    isVisible: {
+      true: {
+        container: "block",
+      },
+      false: {
+        container: "hidden",
+      },
+    },
+  },
+})
+
+const compoundStyles = createImagePreviewStyles()
+
+export const ImageGalleryClient = ({
+  images,
+  assetsBaseUrl,
+  shouldLazyLoad,
+}: ImageGalleryClientProps) => {
+  const [currentIndex, setCurrentIndex] = useState(0)
+  const [isTransitioning, setIsTransitioning] = useState(false)
+
+  const isDesktop = useBreakpoint("lg")
+  const maxPreviewImages = useMemo(() => (isDesktop ? 5 : 3), [isDesktop])
+
+  const containerRef = useRef<HTMLDivElement>(null)
+
+  const previewIndices = getPreviewIndices({
+    numberOfImages: images.length,
+    currentIndex,
+    maxPreviewImages,
+  })
+
+  const preloadImage = useCallback(
+    (index: number) => {
+      // Out of bounds
+      if (index < 0 || index >= images.length) return
+
+      // If the image is the current image, we don't need to preload it
+      if (index === currentIndex) return
+
+      const image = containerRef.current?.querySelectorAll("img")[index]
+      if (!image) return // should never happen since we render all images
+
+      // Image has already been loaded
+      if (image.complete) return
+
+      // Image not loaded yet, force load it!
+      if (image.getAttribute("loading") === "lazy") {
+        image.removeAttribute("loading")
+        const src = image.src
+        image.src = ""
+        image.src = src
+      }
```

Comment:
[comment] not sure if it's possible to do, but the lazy load appears to be a function of the current index and the `maxPreviewImages` - should be just pass in the logic here to the `lazyLoad` prop on `ImageClient`? feels abit clearer and saves us the trouble of dom manipulation

Reply:
not sure if i understand correctly. `maxPreviewImages` is determined by the breakpoints and not a config, and we will only know when the component is rendering